packages/gatsby/src/schema/add-field-resolvers.js

```diff
@@ -0,0 +1,84 @@
+const _ = require(`lodash`)
+const { defaultFieldResolver } = require(`graphql`)
+const { dateResolver } = require(`./types/date`)
+const { link, fileByPath } = require(`./resolvers`)
+
+export const addFieldResolvers = ({
+  schemaComposer,
+  typeComposer,
+  parentSpan,
+}) => {
+  typeComposer.getFieldNames().forEach(fieldName => {
+    let field = typeComposer.getField(fieldName)
+
+    const extensions = typeComposer.getFieldExtensions(fieldName)
+    if (
+      !field.resolve &&
+      extensions.addResolver &&
+      _.isObject(extensions.addResolver)
+    ) {
+      const options = extensions.addResolver.options || {}
+      switch (extensions.addResolver.type) {
+        case `date`: {
+          addDateResolver({
+            typeComposer,
+            fieldName,
+            options,
+          })
+          break
+        }
+        case `link`: {
+          typeComposer.extendField(fieldName, {
```

Comment:
Just to quickly confirm: we only allow one extension per field, so it's ok to `extendField`, otherwise we'd have to wrap the original resolver.

Reply:
Yeah!